src/math/exp_fcn_hifi.c

```diff
@@ -277,10 +280,72 @@ int32_t sofm_exp_int32(int32_t x)
 	return AE_MOVAD32_L(AE_MOVINT32X2_FROMINT64(ts));
 }
 
-#define ONE_Q20         Q_CONVERT_FLOAT(1.0, 20)	  /* Use Q12.20 */
-#define TWO_Q27         Q_CONVERT_FLOAT(2.0, 27)	  /* Use Q5.27 */
-#define MINUS_TWO_Q27   Q_CONVERT_FLOAT(-2.0, 27)	  /* Use Q5.27 */
-#define q_mult(a, b, qa, qb, qy) ((int32_t)Q_MULTSR_32X32((int64_t)(a), b, qa, qb, qy))
+/* Xtensa C macros to function sing HiFi intrinsics*/
+/* Convert a float number to fractional Qnx.ny format. Note that there is no
+ * check for nx+ny number of bits to fit the word length of int. The parameter
+ * qy must be 31 or less.
+ */
+static inline int exp_hifi_q_convert_float(float a, int b)
+{
+	int val;
+	float x, y;
+
+	val = AE_SLAA32(1, b);
+	x = XT_FLOAT_S(val, 0);
+	y = XT_MUL_S(a, x);
+
+	return XT_TRUNC_S(y, 0);
+}
+
+/* Fractional multiplication with shift and round
+ * Note that the parameters px and py must be cast to (int64_t) if other type.
+ */
+static inline int exp_hifi_q_multsr_32x32(int a, int b, int c, int d, int e)
```

Comment:
This is also suspicious, there's no need for generic C speed optimization with generic fractions. You should do this with inline instrinsics with minimal operations if need fast intrinsic multiply. It's would be usually just an AE multiply instruction with a literal constant shift to adjust the fractional format. Please leave optimizations like this to other PR where we can review them more carefully.